src/libraries/System.Threading.Channels/src/System/Threading/Channels/BoundedChannel.cs

```diff
@@ -283,6 +287,23 @@ internal BoundedChannelWriter(BoundedChannel<T> parent)
                 _waiterSingleton = new AsyncOperation<bool>(runContinuationsAsynchronously: true, pooled: true);
             }
 
+            private struct DropItem
+            {
+                internal T Item { get; private set; }
+                internal bool HasValue { get; private set; }
+
+                internal static readonly DropItem None = new DropItem { HasValue = false };
+
+                internal static DropItem Create(T itemToDrop)
+                {
+                    return new DropItem
+                    {
+                        HasValue = true,
+                        Item = itemToDrop,
+                    };
+                }
+            }
```

Comment:
This struct shouldn't be necessary.  You can just have a local bool and a local T.  Adding such a struct just adds additional pressure on the JIT.

Reply:
Removed